tests/contrib/httplib/test_httplib.py

```diff
@@ -186,10 +187,19 @@ def test_httplib_request_get_request_https(self):
         # TODO: figure out how to use https in our local httpbin container
         conn = self.get_https_connection("httpbin.org")
         with contextlib.closing(conn):
-            conn.request("GET", "/status/200")
-            resp = conn.getresponse()
-            self.assertEqual(self.to_str(resp.read()), "")
-            self.assertEqual(resp.status, 200)
+            # Workaround for directly calling httpbin.org is flaky, so try multiple times to get proper 200 response
+            for i in range(50):
+                try:
+                    conn.request("GET", "/status/200")
+                    resp = conn.getresponse()
+                    if resp.status == 200:
+                        self.assertEqual(self.to_str(resp.read()), "")
+                        break
+                except Exception:
```

Comment:
can we catch the specific exception here?

if an unexpected exception happens we should just fail.

Reply:
This is a weird try/catch block. Essentially we have a check looking for a 200 status code, the assertion inside that block should always pass if there is a status 200. I'll change this around so it makes more sense.